test/e2e/kubernetes/workloads/reboot-control-plane-node.yaml

```diff
@@ -0,0 +1,111 @@
+# How this works:
+# On every master, a pod will run that will try to get an annotation lock
+# once it gets the lock for its node, it releases it, sleeps a bit and
+# then loops until it can get it again.
+# After it gets it the second time, it will then trigger a node hard reboot
+# (Since force or double force)
+# Then, when the node (master) comes back up, it will again run this pod
+# and it will again see it has the annotation and will release it, letting
+# another node race to get it.
+# Works with ubuntu 16.04 and 18.04 on kubernetes 1.15+  (may work earlier)
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  labels:
+    app: reboot-test
+  name: reboot-test
+  namespace: default
+spec:
+  selector:
+    matchLabels:
+      app: reboot-test
+  template:
+    metadata:
+      labels:
+        app: reboot-test
+    spec:
+      # This restricts this to masters
+      affinity:
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+              - matchExpressions:
+                  - key: kubernetes.azure.com/role
+                    operator: In
+                    values:
+                      - master
+      containers:
+        - name: reboot-test
+          image: ubuntu:18.04
+          imagePullPolicy: IfNotPresent
+          command:
+            - bash
+            - -c
+            - >-
+              while [[ $(kubectl annotate namespace ${LOCK_NS} ${LOCK_NAME}=${NODE_ID} 2>&1) != *\(${NODE_ID}\)* ]];

```

Comment:
Minor addition - all of the kubectl command should also have --request-timeout 30s  (or pick your number) since the reboots of another node may hang the kubectl command otherwise.

For example:
`kubectl --request-timeout 30s annotate namespace ${LOCK_NS} ${LOCK_NAME}=${NODE_ID}`